php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -896,4 +895,24 @@ public function byteSize()
         }
         return $size;
     }
+
+    private function appendHelper($field, $append_value)
+    {
+        $getter = $field->getGetter();
+        $setter = $field->getSetter();
+
+        $field_arr_value = $this->$getter();
+        $field_arr_value[] = $append_value;
+        $this->$setter($field_arr_value);
+    }
+
+    private function kvUpdateHelper($field, $update_key, $update_value)
+    {
+        $getter = $field->getGetter();
+        $setter = $field->getSetter();
+
+        $field_arr_value = $this->$getter();
+        $field_arr_value[$update_key] = $update_value;
+        $this->$setter($field_arr_value);
```

Comment:
Yes, on line 914, (internally in PHP) we get a reference to the array, then on line 915 PHP does a CoW and we update the local copy of the array, then on 916 we update the original source of the array with the updated local copy. This is likely what was happening also in the previous syntax; it's just not compatible in HHVM because the behavior with the old syntax is ambiguous with PHP's CoW system.